app/puppet-access/login.go

```diff
@@ -6,13 +6,52 @@ import (
 
 	"github.com/puppetlabs/pe-sdk-go/app/puppet-access/api/client/operations"
 	"github.com/puppetlabs/pe-sdk-go/app/puppet-access/api/models"
+	"github.com/puppetlabs/pe-sdk-go/log"
 )
 
+func errorHandling(err error) error {
+	if du, ok := err.(*operations.LoginUnauthorized); ok {
+		if du.Payload != nil {
+			log.Debug(err.Error())
+			return fmt.Errorf("Recieved an error while trying to request the token. The error was %v: %v", du.Payload.Kind, du.Payload.Msg)
+		}
+	}
+	if du, ok := err.(*operations.LoginBadRequest); ok {
+		if du.Payload != nil {
+			log.Debug(err.Error())
+			return fmt.Errorf("Recieved an error while trying to request the token. The error was %v: %v", du.Payload.Kind, du.Payload.Msg)
+		}
+	}
+	if du, ok := err.(*operations.LoginDefault); ok {
+		log.Debug(err.Error())
+		return fmt.Errorf("Recieved an error while trying to request the token. The error with status code %d was %v", du.Code(), du.Payload)
+	}
+
+	log.Debug(err.Error())
+	return fmt.Errorf("Unkown error")
+
+}
+
+//LoginWithErrorDetails ...
+func (puppetAccess *PuppetAccess) LoginWithErrorDetails() (string, error) {
+	resp, err := puppetAccess.Login()
+	if err != nil {
+		log.Debug(err.Error())
+		err := errorHandling(err)
+		return "", err
+	}
+	if resp.Payload.Token == "" {
+		return "", fmt.Errorf("The response did not contain a token. Rerun with --debug to see full body")
+	}
+
+	return resp.Payload.Token, nil
+}
+
 // Login method requests a token from the server
-func (puppetAccess *PuppetAccess) Login() (string, error) {
+func (puppetAccess *PuppetAccess) Login() (*operations.LoginOK, error) {
 	client, err := puppetAccess.Client.GetClient()
```

Comment:
this should be moved to `LoginWithErrorDetails` and `Login` method could be removed